internal/statesync/reactor.go

```diff
@@ -102,10 +102,21 @@ const (
 	maxLightBlockRequestRetries = 20
 )
 
-// SyncReactor defines an interface used for testing abilities of node.startStateSync.
+// SyncReactor defines an interface used for testing abilities of node.startStateSync
+// or the rpc syncinfo query.
 type SyncReactor interface {
 	Sync(context.Context, StateProvider, time.Duration) (sm.State, error)
 	Backfill(sm.State) error
+
+	// The following calls are for the rpc sync info query, please see statesync.metrics for
+	// the details.
+	GetTotalSnapshots() int64
```

Comment:
This is a pretty substantial expansion of this interface. Would it make more sense to define this as a separate interface that the reactor can return? E.g., 

```go
type SyncReactor interface {
  Sync(…)
  Backfill(…) error
  Metrics() Metrics
}

type Metrics interface {
   GetTotalSnapshots() int64
   // …
}
```

That could also be helpful for implementations that don't provide metrics—the API can allow a `nil` collector to stand for no-op without having to export that explicitly.

Reply:
Do we want to return different rpc syncInfo responses base on the metrics setup? @cmwaters?

Reply:
Separated APIs to the `Metricer` interface.

Reply:
> Do we want to return different rpc syncInfo responses base on the metrics setup? 

No, I think having a single syncInfo response is sufficient